src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1505,6 +1507,39 @@ public boolean checkIfRequestForItemExistOrCreateAsynch(@NotNull final ItemStack
         return false;
     }
 
+    /**
+     * Check if a tag has been requested already or is in the inventory. If not in the inventory and not requested already, create request
+     *
+     * @param tag the requested tag.
+     * @return true if in the inventory, else false.
+     */
+    public boolean checkIfRequestForTagExistOrCreateAsynch(@NotNull final Tag<Item> tag, final int count)
+    {
+        if (InventoryUtils.hasItemInItemHandler(worker.getInventoryCitizen(), stack -> stack.getItem().isIn(tag) && stack.getCount() >= count))
+        {
+            return true;
+        }
+
+        if (InventoryUtils.getItemCountInProvider(getOwnBuilding(),
+          itemStack -> itemStack.getItem().isIn(tag)) >= count &&
+              InventoryUtils.transferXOfFirstSlotInProviderWithIntoNextFreeSlotInItemHandler(
+                getOwnBuilding(), itemStack -> itemStack.getItem().isIn(tag),
+                        count,
+                        worker.getInventoryCitizen()))
+        {
+            return true;
+        }
+
+        if (getOwnBuilding().getOpenRequestsOfTypeFiltered(worker.getCitizenData(), TypeConstants.DELIVERABLE,
```

Comment:
You can here filter it by TagRequest instead of Deliverable already, that's what I meant

Reply:
So I should create a new entry in TypeConstants

Reply:
yes